com.unity.netcode.gameobjects/Runtime/SceneManagement/NetworkSceneManager.cs

```diff
@@ -224,16 +240,75 @@ public class NetworkSceneManager : IDisposable
         /// </summary>
         public void Dispose()
         {
-            SceneEventData.Dispose();
-            SceneEventData = null;
+            foreach (var sceneEventData in SceneEventDataPool)
+            {
+                sceneEventData.Dispose();
+            }
+            SceneEventDataPool.Clear();
+            SceneEventDataPool = null;
             ClientSynchEventData.Dispose();
             ClientSynchEventData = null;
         }
 
         /// <summary>
-        /// Gets the scene name from full path to the scene
+        /// Depending upon how many users are connected, you might need to adjust
+        /// the <see cref="SceneEventData"/> pool size to a larger value on a host
+        /// or a server under the situation where you are loading and/or unloading
+        /// many additive scenes over a short period of time.
+        /// The default <see cref="SceneEventData"/> pool size is 10 (<see cref="DefaultSceneEventDataPoolSize"/>)
+        /// The maximum <see cref="SceneEventData"/> pool size can grow to is 100 (<see cref="MaximumSceneEventDataPoolSizeThreshold"/>)
         /// </summary>
+        /// <param name="sceneEventDataPoolSize"></param>
         /// <returns></returns>
+        public bool SetSceneEventDataPoolSize(int sceneEventDataPoolSize)
```

Comment:
Connected to our previous conversation, this seems like a fair bit of complexity and configurability for something that I don't think users will get into.

What I see here is, we have 3 settings which 
- if exceeded generate a warning
- but, the situation is worse than a warning - when the warning happens it actually has failed to do the operation.  That sounds more like an error to me.  Or somehow is the pool failing to grow still ok?

Again I think these settings add complexity and barring coming to an improved understanding would like to remove them.

Put another way, if your game is set up such that you blow through our expected maximums, just having a "you blew through our maximums so set them higher else we'll warn you and put you in an unexpected state" is not adding value.

If you want to convince me that we must give them a way to increase DefaultSceneEventDataPoolSize because whatever default we pick would if hard-coded cause harm to users because they'll have to waste cycles in their pool growing up to their larger amounts, we can have that conversation too.

Reply:
This was resolved by just dynamically generating SceneEventData objects per each unique scene event.